x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/EsQueryExec.java

```diff
@@ -56,8 +59,19 @@ public class EsQueryExec extends LeafExec implements EstimatesRowSize {
      */
     private final Integer estimatedRowSize;
 
-    public record FieldSort(FieldAttribute field, Order.OrderDirection direction, Order.NullsPosition nulls) implements Writeable {
-        public FieldSortBuilder fieldSortBuilder() {
+    public interface Sort {
+        SortBuilder<?> sortBuilder();
```

Comment:
I think we should not be using DSL objects here. Could not we use here a lucene SortField instead?

Reply:
As I understand it we'll consider that later, so I'll mark this as resolved.